video-presentation.py

```diff
@@ -230,6 +233,43 @@ def parsevalue(size, max_size):
         self.logger.debug("Masks after preparing: %s", parsed_masks)
         return parsed_masks
 
+    def create_video_previews(self, filename):
+        """Create mp4 and webm heavily compressed previews of the presentation to use in the previewer"""
+
+        # Let's not be greedy, use half available cores since we are probably in a docker container
+        encoding_threads = multiprocessing.cpu_count()
+        if encoding_threads > 1:
+            encoding_threads = int(np.ceil(encoding_threads/2))
+
+        ffmpeg_stub = "ffmpeg -loglevel error -y -i \"" + os.path.abspath(filename) + "\" -threads " + \
+                      str(encoding_threads)
+        # We use the same audio settings for both videos
+        no_audio = " -an "
+        audio = " -acodec libvorbis -b:a 64k "
+        # Use very heavy compression since most of what we deal with is 2d without shadows
+        mp4_settings = " -vcodec libx264 -preset medium -b:v 96k -qmax 42 -maxrate 250k "
+        webm_settings = " -vcodec libvpx -quality good -b:v 96k -crf 10 -qmin 0 -qmax 42 -maxrate 250k -bufsize 1000k "
+
+        # The 2 pass method requires some temporary files so let's (temporarily) change to the temp dir we have
+        currentdir = os.getcwd()
+        os.chdir(self.tempdir)
+
+        # First let's do mp4
+        ffmpeg_command = ffmpeg_stub + mp4_settings + no_audio + "-pass 1 -f mp4 /dev/null"
+        subprocess.check_output(ffmpeg_command, stderr=subprocess.STDOUT, shell=True)
+        ffmpeg_command = ffmpeg_stub + mp4_settings + audio + "-pass 2 -f mp4 preview.mp4.preview"
+        subprocess.check_output(ffmpeg_command, stderr=subprocess.STDOUT, shell=True)
+        # Now do webm
+        ffmpeg_command = ffmpeg_stub + webm_settings + no_audio + "-pass 1 -f webm /dev/null"
+        subprocess.check_output(ffmpeg_command, stderr=subprocess.STDOUT, shell=True)
+        ffmpeg_command = ffmpeg_stub + webm_settings + audio + "-pass 2 -f webm preview.webm.preview"
+        subprocess.check_output(ffmpeg_command, stderr=subprocess.STDOUT, shell=True)
```

Comment:
there is no error handling? At least return code and a `try except` block should be required?

Reply:
Of course you are right...but for some reason with these settings if the command fails, the failure is actually communicated back to Clowder and printed in the extraction events.